### PR TITLE
My Home: Combine photo library buttons to one support dialog and update image.

### PR DESCRIPTION
**Pull request by kwight**
For the Free Photos card, we currently have an image that opens a dialog with just the video, and a Learn More button that opens a dialog displaying the support page related to free photos. I've added the video to the [support page](https://wordpress.com/support/free-photo-library/), which now means the video displays at the top of the support dialog – kinda making the separate dialog for the video redundant. This PR has clicking on the image open the same support dialog as the Learn More button, meaning whichever the user clicks on, they'll get both sources of support (with the video being first in sight).

This will also make the (not autoplay) video available to users on mobile, since the card image is not displayed at smaller screen widths (only the Learn More button).

Screen Shot 2020-03-18 at 11 47 19 AM

#### Testing instructions

* On this branch, go to `/home/:site`.
* Verify that clicking the image in the Free Photos card opens the support dialog, and that the Learn More button continues to do the same. The video should be the first content in the support dialog, after the page title.

### CODE REVIEW

**Comment by gwwar** [comment]
Hmm, won't it be difficult to track intent to watch the video? I suppose we could rely on YouTube analytics only, but this might make generation of funnels more difficult

**Comment by kwight** [comment]
Opening the dialog will trigger a console error:

Screen Shot 2020-03-18 at 11 53 57 AM

I'm not clear how much of a problem this is.

**Comment by gwwar** [comment]
Is the error on the iframe'd support page?

**Comment by gwwar** [comment]
One other danger is that the support page may be updated without the video being at the top of the page. Thoughts here @danhauk @jancavan @sixhours ?

File: client/my-sites/customer-home/free-photo-library-card/index.jsx [hunk 1]
```diff
@@ -29,65 +29,37 @@ const FreePhotoLibraryCard = ( {
...
-		if ( ! showDialog ) {
-			tracks( 'calypso_customer_home_free_photo_library_video_dialog_view' );
-		}
-		return setShowDialog( ! showDialog );
+	const displayDialog = () => {
+		tracks( 'calypso_customer_home_free_photo_library_video_support_page_view' );
+		supportArticleDialog( {
+			postId: 145498,
+			postUrl: localizeUrl( 'https://support.wordpress.com/free-photo-library/' ),
+		} );
```
Comment (gwwar) [inline client/my-sites/customer-home/free-photo-library-card/index.jsx]: see https://github.com/Automattic/wp-calypso/pull/40258

**Comment by gwwar** [comment]
Thanks for updating the support page @kwight! I'll defer to design here for behavior

**Comment by kwight** [comment]
> Hmm, won't it be difficult to track intent to watch the video?

831161fcb8cfcd0e2120e8c3e48282a0830225b7 is functionally the same for tracking as before, while maintaining the better user behaviour.

**Comment by gwwar** [comment]
My janitorial will likely conflict with this patch, do you mind if we land that first so this one can be rebased?

I do want to tidy that action thunk that doesn't need to exist.

**Comment by mmtr** [comment]
Should we remove the "play" icon from the thumbnail? I guess users will expect a video being autoplayed otherwise after clicking on the thumbnail.

Screen Shot 2020-03-19 at 12 14 48

**Comment by danhauk** [comment]
> One other danger is that the support page may be updated without the video being at the top of the page.

That's a slight risk, but I think it's ok. In other support pages with videos I've seen, the video is also at the top so it seems to be an established pattern already. And if we work directly with marketing and HEs to make sure that's communicated we should be ok.

> Should we remove the "play" icon from the thumbnail?

@mmtr I think the play icon is ok. It's a nice indicator that there's a video. Without it, there's an expectation that it's something I'm going to have to read.

That being said, I'd love to update the thumbnail to something other than a blurry screenshot (not a blocker for this PR). Thoughts on something like this, and moving the video indication to the button? cc @sixhours @jancavan 

Group 218

**Comment by sixhours** [comment]
> That being said, I'd love to update the thumbnail to something other than a blurry screenshot (not a blocker for this PR). 

Yes, I think that's a great idea. A graphic might be less busy, but a non-blurry photo would look better, too.

**Comment by danhauk** [comment]
After a discussion with @sixhours we decided to go with the following illustration for a few reasons:

* It mimics the Gutenberg illustration style in the welcome guide which helps to start unifying things between managing and editing.
* Maintenance is difficult for UI screenshots - every time an update is made we need to audit screenshots.
* This illustration gives some context of where to find the photo library without being literal.

image

**Comment by jancavan** [comment]
I like that "Learn more" button with a filmstrip icon!

I still like showing the real photos though because it showcases the free photo library even more. IMO, it doesn't really imply that it's an auto-playing video because it doesn't actually auto-play? Or I may be misunderstanding it.

**Comment by gwwar** [comment]
😆 If we're really unsure about the image, we can always A/B test which image works better. (A proper good use case of A/B testing). Overall I think removing the play icon from the thumbnail is fine. More folks are clicking on the button CTA anyway.

File: client/my-sites/customer-home/free-photo-library-card/index.jsx [hunk 2]
```diff
@@ -23,71 +23,49 @@ import './style.scss';
...
-	const [ showDialog, setShowDialog ] = useState( false );
 	const translate = useTranslate();
 
-	const toggleDialog = () => {
-		if ( ! showDialog ) {
+	const displayDialog = clickSource => {
+		if ( clickSource === 'prompt' ) {
 			tracks( 'calypso_customer_home_free_photo_library_video_dialog_view' );
+		} else {
+			tracks( 'calypso_customer_home_free_photo_library_video_support_page_view' );
```
Comment (gwwar) [inline client/my-sites/customer-home/free-photo-library-card/index.jsx]: Non-blocking, but I noticed while refactoring quick links that we're also bumping stats under the 'calypso_customer_home' link. Might be worth adding something like this this
```jsx
composeAnalytics(
    bumpStat( 'calypso_customer_home', 'view_free_photo_library_video' ),
    tracks( 'calypso_customer_home_free_photo_library_video_support_page_view' ),
)
```

Comment (kwight) [reply client/my-sites/customer-home/free-photo-library-card/index.jsx]: ff69cfc1f14fb9a72e1a743dadd1d14444cf3574

**Pull request merged**